CMakeLists.txt

```
@@ -0,0 +1,169 @@
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+
+project(KokkosTools CXX)
+
+# Set policies
+cmake_policy(SET CMP0054 NEW) # simplify if()
+cmake_policy(SET CMP0057 NEW) # support IN_LIST in if()
+cmake_policy(SET CMP0111 NEW) # error if library not found
+
+# Disable in-source builds to prevent source tree corruption.
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
+  message(FATAL_ERROR "FATAL: In-source builds are not allowed. You should create a separate directory for build files.")
+endif()
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
```

```suggestion
list(INSERT CMAKE_MODULE_PATH 0 ${PROJECT_SOURCE_DIR}/cmake)
```

You should ensure that your `FindPAPI.cmake` is always picked up first.